test/unit/factorization/mc/test_compute_t_factor.cpp

```diff
@@ -0,0 +1,270 @@
+// can use to compute the Householder transformation by applying all reflectors in block.
+// This Householder transformation is obtained with the equation
+//
+// H = I - V . T . V*
+//
+// Which we expect to be the equal to the one computed previously.
+TYPED_TEST(ComputeTFactorDistributedTest, Correctness) {
+  using namespace dlaf;
+
+  constexpr auto error = test::TypeUtilities<TypeParam>::error;
+
+  SizeType a_m, a_n, mb, nb, k;
+
+  for (auto comm_grid : this->commGrids()) {
+    for (auto config : configs) {
+      std::tie(a_m, a_n, mb, nb, k) = config;
+
+      const TileElementSize block_size(mb, nb);
+
+      comm::Index2D source_rank_index(std::max(0, comm_grid.size().rows() - 1),
+                                      std::min(1, comm_grid.size().cols() - 1));
+      const matrix::Distribution dist_v({a_m, a_n}, block_size, comm_grid.size(), comm_grid.rank(),
+                                        source_rank_index);
+
+      const GlobalTileIndex v_start{dist_v.nrTiles().rows() / 2, dist_v.nrTiles().cols() / 2};
+
+      Matrix<const TypeParam, Device::CPU> v_input = [&]() {
+        Matrix<TypeParam, Device::CPU> V(dist_v);
+        dlaf::matrix::util::set_random(V);
+        return V;
+      }();
+
+      const MatrixLocal<const TypeParam> v = [&v_input, &dist_v, &comm_grid, a_m, a_n, nb, v_start] {
+        // TODO this can be improved by communicating just the interesting part
+        // gather the entire A matrix
+        auto a = matrix::test::all_gather<TypeParam>(v_input, comm_grid);
+
+        // panel shape
+        const auto v_start_el = dist_v.globalElementIndex(v_start, {0, 0});
+        const auto v_end_el = GlobalElementIndex{a_m, std::min(v_start_el.col() + nb, a_n)};
+        const auto v_size_el = v_end_el - v_start_el;
+
+        MatrixLocal<TypeParam> v(v_size_el, a.blockSize());
+        DLAF_ASSERT_HEAVY(v.size().rows() > v.size().cols(), v.size());
+
+        // copy only the panel
+        const GlobalTileSize v_offset{v_start.row(), v_start.col()};
+        for (const auto& ij : iterate_range2d(v.nrTiles()))
+          copy(a.tile_read(ij + v_offset), v.tile(ij));
+
+        // clean reflectors
+        // clang-format off
+        lapack::laset(lapack::MatrixType::Upper,
+            v.size().rows(), v.size().cols(),
+            0, 1,
+            v.ptr(), v.ld());
+        // clang-format on
+
+        return v;
+      }();
+
+      const SizeType m = v.size().rows();
+
+      // compute taus and H_exp
+      common::internal::vector<hpx::shared_future<TypeParam>> taus_input;
+      taus_input.reserve(k);
+
+      MatrixLocal<TypeParam> h_expected({m, m}, block_size);
+      set(h_expected, preset_eye<TypeParam>);
+
+      for (auto j = 0; j < k; ++j) {
+        const SizeType reflector_size = m - j;
+
+        const TypeParam* data_ptr = v.ptr({j, j});
+        const auto norm = blas::nrm2(reflector_size, data_ptr, 1);
+        const TypeParam tau = 2 / std::pow(norm, 2);
```

Comment:
After an offline discussion with @rasolca, it has been decided that it may be better to leave it as it is with `nrm2+pow` instead of `dot`.

The main point could be the accuracy of the results, since `norm2` internally ensures that an overflow does not occur (if I get it correctly).